app/gen-server/ApiServer.ts

```diff
@@ -627,6 +636,147 @@ export class ApiServer {
+            ...( payload.expiresAt ? { expiresAt: new Date(payload.expiresAt) } : {} ),
+            expiresAt: payload.expiresAt !== undefined ? new Date(payload.expiresAt) : undefined,
+          };
+
+          const resp = await this._dbManager.updateServiceAccount(
+            serviceAccountId, updateProps, { expectedOwnerId: userId }
+          );
+          if (!resp) {
+            throw new ApiError(`No such service account as "${serviceAccountId}"`, 404);
+          }
+          return sendOkReply(req, res, resp);
+        })
+      );
+
+      // DELETE /service-accounts/:said
+      // Deletes one particular service account of the user making the api call.
+      this._app.delete('/api/service-accounts/:said', expressWrap(async (req, res) => {
+        const userId = getAuthorizedUserId(req);
+        const serviceAccountId = parseInt(req.params.said);
+        const resp = await this._dbManager.deleteServiceAccount(serviceAccountId, {expectedOwnerId: userId});
+        if (resp === null) {
+          throw new ApiError(`No such service account as "${serviceAccountId}"`, 404);
+        }
+        return sendOkReply(req, res, resp);
+      }));
+
+      // POST /service-accounts/:said/apikey
+      // Regenerate and return the apikey of a given Service Account
+      this._app.post('/api/service-accounts/:said/apikey', expressWrap(async (req, res) => {
+        const userId = getAuthorizedUserId(req);
+        const serviceAccountId = parseInt(req.params.said);
+        const serviceAccount = await this._dbManager.createServiceAccountApiKey(
+          serviceAccountId, {expectedOwnerId: userId}
+        );
+        if (serviceAccount === null) {
+          throw new ApiError(
+            `Can't regenerate api key of non existing service account ${serviceAccountId}`,
+            404
+          );
+        }
+        const resp: SATypes.ServiceAccountApiResponse = {
+          id: serviceAccount.id,
+          key: serviceAccount.serviceUser.apiKey,
+          label: serviceAccount.label,
+          description: serviceAccount.description,
+          expiresAt: serviceAccount.expiresAt.toISOString(),
+          hasValidKey: true
+        };
+        return sendOkReply(req, res, resp);
+      }));
+
+      // DELETE /service-accounts/:said/apikey
+      // Deletes the apikey of a given Service Account by deleting the key
+      this._app.delete('/api/service-accounts/:said/apikey', expressWrap(async (req, res) => {
+        const userId = getAuthorizedUserId(req);
+        const serviceAccountId = parseInt(req.params.said);
+        const serviceAccount = await this._dbManager.deleteServiceAccountApiKey(
+          serviceAccountId, {expectedOwnerId: userId}
+        );
+        if (serviceAccount == null) {
+          throw new ApiError(
+            `Can't delete api key of non existing service account ${serviceAccountId}`,
+            404
+          );
+        }
```

Comment:
This should be unnecessary, `deleteServiceAccountApiKey` normally raises an error if not found.